src/modules/launcher/Plugins/Microsoft.PowerToys.Run.Plugin.TimeZone/Classes/TimeZoneSettings.cs

```diff
@@ -0,0 +1,93 @@
+        /// Gets or sets a value indicating whether the time zone name of a time zone is shown in the results.
+        /// </summary>
+        internal bool ShowTimeZoneNames { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the time name of a time zone is shown in the results.
+        /// </summary>
+        internal bool ShowTimeNames { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value indicating whether the military name of a time zone is shown in the results.
+        /// </summary>
+        internal bool ShowMilitaryTimeZoneNames { get; set; }
+
+        /// <summary>
+        /// Return a list with all settings. Additional
+        /// </summary>
+        /// <returns>A list with all settings.</returns>
+        internal List<PluginAdditionalOption> GetAdditionalOptions()
+        {
+            var optionList = new List<PluginAdditionalOption>
+            {
+                new PluginAdditionalOption
+                {
+                    Key = nameof(ShowTimeZoneNames),
+                    DisplayLabel = Resources.ShowTimeZoneNames,
+                    Value = true,
+                },
+                new PluginAdditionalOption
+                {
+                    Key = nameof(ShowTimeNames),
+                    DisplayLabel = Resources.ShowTimeNames,
+                    Value = true,
+                },
+                new PluginAdditionalOption
+                {
+                    Key = nameof(ShowMilitaryTimeZoneNames),
+                    DisplayLabel = Resources.ShowMilitaryTimeZoneNames,
+                    Value = false,
+                },
+            };
+
+            return optionList;
+        }
+
+        /// <summary>
+        /// Update this settings.
+        /// </summary>
+        /// <param name="settings">The settings for all power launcher plugin.</param>
+        internal void UpdateSettings(PowerLauncherPluginSettings settings)
+        {
+            if (settings is null || settings.AdditionalOptions is null)
+            {
+                return;
+            }
+
+            ShowTimeZoneNames = GetOneBoolSetting(settings, nameof(ShowTimeZoneNames), true);
+            ShowTimeNames = GetOneBoolSetting(settings, nameof(ShowTimeNames), true);
+            ShowMilitaryTimeZoneNames = GetOneBoolSetting(settings, nameof(ShowMilitaryTimeZoneNames), false);
+        }
+
+        /// <summary>
+        /// Return one <see cref="bool"/> setting of the given settings list with the given name.
+        /// </summary>
+        /// <param name="settings">The object that contain all settings.</param>
+        /// <param name="name">The name of the setting.</param>
+        /// <param name="fallbackValue">The fall-back value that is used when the setting is not found.</param>
+        /// <returns>A settings value.</returns>
+        private static bool GetOneBoolSetting(PowerLauncherPluginSettings settings, string name, bool fallbackValue)
```

Comment:
`GetSettingsByName`?

Reply:
Next time, we should convert this to a extension Method, introduce a new Interface for plugin settings and put this into the runner, to reduce copy&paste code, something like this
`public static T GetSettingOrDefault<T>(IPluginSettings setting, string name, T defaultValue)`.

Reply:
I think we should move the whole update logic out of the plugins. So that we only define `PluginAdditionalOptions` and can access the settings value (or the default one) from `PowerLauncherSettings.PluginOptions.GetForPlugin` or something.